### ISSUE TITLE
Expression bug when updating

### ISSUE THREAD

**Issue by Centurys** [linq2db/linq2db#1554]
I have DB model class with fluent expression defined. One property "ClaimedKeyType" is enum type.
After update to 2.6.3 from 2.6.2 I got this exception when tying to use query.Set().Set().Update() api's.

```
System.ArgumentException: Property '*.KeyTypes ClaimedKeyType' is not defined for type 'System.Object'
   at System.Linq.Expressions.Expression.Property(Expression expression, PropertyInfo property)
   at System.Linq.Expressions.Expression.MakeMemberAccess(Expression expression, MemberInfo member)
   at LinqToDB.Linq.Builder.UpdateBuilder.ParseSet(ExpressionBuilder builder, BuildInfo buildInfo, LambdaExpression extract, Expression update, IBuildContext select, List`1 items)
   at LinqToDB.Linq.Builder.UpdateBuilder.Set.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.UpdateBuilder.Set.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.UpdateBuilder.Set.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.UpdateBuilder.Set.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.UpdateBuilder.Set.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.UpdateBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.Build[T]()
   at LinqToDB.Linq.Query`1.CreateQuery(IDataContext dataContext, Expression expr)
   at LinqToDB.Linq.Query`1.GetQuery(IDataContext dataContext, Expression& expr)
   at LinqToDB.Linq.ExpressionQuery`1.GetQuery(Expression& expression, Boolean cache)
   at LinqToDB.Linq.ExpressionQuery`1.System.Linq.IQueryProvider.Execute[TResult](Expression expression)
   at LinqToDB.LinqExtensions.Update[T](IUpdatable`1 source)
```

### Environment details
linq2db version: 2.6.3
Database Server: MSSQL 2016
Operating system: Windows
Framework version: *.NET Core 2.2

**Comment by sdanyliv** [comment]
@Centurys, sorry about that, looks like regression.
Could you post model and linq Update expression?

**Comment by Centurys** [comment]
It's not full code, because there are to many text to copy, but I hope it's mots important part to reproduce.

```c#
    // Model
    public class PersonCache
    {
        public int Id { get; set; }

        public string PhoneNumber { get; set; }

        public KeyTypes ClaimedKeyType { get; set; }

        public DateTimeOffset Updated { get; set; }
    }

    [Flags]
    public enum KeyTypes : byte
    {
        RSA = 1,
        EC = 2
    }

    // Fluent
    MappingSchema.GetFluentMappingBuilder().Entity<PersonCache>()
                .HasTableName("PersonCaches")
                .Property(p => p.Id)
                    .IsPrimaryKey()
                    .IsIdentity()
                .Property(p => p.PhoneNumber)
                    .IsNullable()
                .Property(p => p.ClaimedKeyType)
                .Property(p => p.Updated);

    // Code
   dataConnection.GetTable<PersonCache>().Where(p => p.Id == 0).AsUpdatable().Set(p => p.ClaimedKeyType, claimedKeyType).Set(p => p.Updated, now).Update()
```

**Comment by sdanyliv** [comment]
@Centurys, unfortunately i can not reproduce bug. Possibly you have missed some important part of your linq query. 
You can check branch  https://github.com/linq2db/linq2db/tree/issue_1554
With test https://github.com/linq2db/linq2db/blob/issue_1554/Tests/Tests.Playground/Issue1554Tests.cs

**Comment by vladimir-kovalyuk** [comment]
We experienced similar issue. The cause was related to building the Func used in Set(p =>p.smth) dynamically via expression tree.

**Comment by sdanyliv** [comment]
@vladimir-kovalyuk, show me the sample and i'll fix that.

**Comment by sdanyliv** [comment]
Do it ASAP, we plan to publish bug-fix release tomorrow.

**Comment by vladimir-kovalyuk** [comment]
We fixed the way we built expression tree. So we no longer experience that issue.

**Comment by MaceWindu** [comment]
We are going to revert this fix in 2.8.1 as it introduced major regression and look into fixing it in later releases

**Comment by MaceWindu** [comment]
fixed by #1846
